xdas/io/febus.py

```diff
@@ -1,27 +1,89 @@
+def read(fname, overlaps=None, offset=None):
+    """
+    Open a Febus file into a xdas DataArray object.
+
+    The Febus file format contains a 3D array which is a stack of 2D (time, distance)
+    chunks of data that overlaps with each other. The overlaps must be trimmed and the
+    chunks concatenated to form a seamless dataset. Each chunk is associated with a
+    timestamp that is located at a fixed offset from the beginning of the chunk.
+
+    Because of poor documentation of the evolution of the Febus file format, it is
+    recommended to manually specify the overlap and offset parameters. If not provided,
+    the function will attempt to determine the correct values at your own risk.
+
+    Parameters:
+    -----------
+    fname : str
+        The filename of the Febus file to read.
+    overlaps : tuple of int, optional
+        A tuple specifying the overlap in number of sample to trim on both side of each
+        chunk of the data. If as scalar is provided, the same overlap will be applied
+        to both sides. If not provided, the function will attempt to determine the
+        correct overlap at your own risk.
+    offset : int, optional
+        The location of the timestamp within each block given as the number of samples
+        from the beginning. If not provided, the function will attempt to determine the
+        correct offset at you own risk.
+
+    Returns:
+    --------
+    DataArray
+        A data array containing the data from the Febus file.
+
+    """
     with h5py.File(fname, "r") as file:
         (device_name,) = list(file.keys())
         source = file[device_name]["Source1"]
         times = np.asarray(source["time"])
         zone = source["Zone1"]
         (name,) = list(zone.keys())
         chunks = VirtualSource(zone[name])
-        delta = [zone.attrs["Spacing"][1] / 1000.0, zone.attrs["Spacing"][0]]
+        delta = (zone.attrs["Spacing"][1] / 1000.0, zone.attrs["Spacing"][0])
     name = "".join(["_" + c.lower() if c.isupper() else c for c in name]).lstrip("_")
-    noverlap = chunks.shape[1] // 4
-    chunks = chunks[:, noverlap:-noverlap, :]
-    times = times + noverlap * delta[0]
+
+    match overlaps:
+        case None:
+            warnings.warn(
+                "No overlap specified, Xdas will try its best to find the correct trimming"
+            )
+            if "BlockOverlap" in zone.attrs:
+                overlap = zone.attrs["BlockOverlap"][0]
+            else:
+                overlap = chunks.shape[1] // 4
+            overlaps = (overlap, overlap)
+        case int(overlap):
+            overlaps = (overlap, overlap)
+        case (int(), int()):
+            pass
+        case _:
+            raise ValueError("overlaps must be a integer or a tuple of two integers")
+
+    match offset:
+        case None:
+            warnings.warn(
+                "No offset specified, Xdas will try its best to place the timestamps"
+            )
+            offset = overlaps[0]
```

Comment:
@ClaudioStrumia Check this please.

Reply:
I think that as it is right now it is not correct beacouse offset = overlap[0] means that then the t0 that we concatenate are referred to the start of the chunk, overlap included. I would change such that if the user specifies no offset, the default is taken to put the t0 in the middle of the time chunk.
If a user has old version Febus data that have no issues relative to the starttime position, we can specify in the warnings that she/he should put offset = (0, 0) so that everything is correct.

Reply:
Yeah maybe using as default the latest version we know about is a good choice.